### PR TITLE
[LUM-723] Fix IBC Barberry patch and kill unused pool with ID #1

### PR DESCRIPTION
**Pull request by Segfaultd**
### Introduction

This PR addresses two points:

- Apply the IBC Barberry patch by bumping relevant dependencies in the go.mod
- Kill the first pool that got improperly created in the v1.4.1 upgrade due to chain halt

On a side note, it also registers the snapshots and pruning command introduced a bit earlier in the Cosmos SDK v0.46 serie.

### Testing

Testing the upgrade handlers is planned on the public Lum Network testnet before releasing the coordinated upgrade.

### CODE REVIEW

**Review by Ricardo-Remy — APPROVED** [review]
LGTM. Let's manage deeper the status change of a Ready to Kill in a next iteration with all the business logic it could have (stop drawing, withdraw etc..)

File: x/millions/keeper/keeper_pool.go [hunk 1]
```diff
@@ -435,6 +435,24 @@ func (k Keeper) UpdatePool(
 	return nil
 }
 
+func (k Keeper) KillPool(ctx sdk.Context, poolID uint64) (types.Pool, error) {
```
Comment (lebascou) [inline x/millions/keeper/keeper_pool.go:438]: Please rename this method as `UnsafeKillPool` and add appropriate comment to mention the temporary and exceptional use of this feature.

**Review by lebascou — CHANGES_REQUESTED** [review]
LGTM - please apply the requested change for the sake of code clarity

**Comment by lebascou** [comment]
> LGTM. Let's manage deeper the status change of a Ready to Kill in a next iteration with all the business logic it could have (stop drawing, withdraw etc..)
Yes ! I checked all the key code components and setting this state in a quite "hardcoded" manner will have the desired effect (prevent deposits and draws on the deprecated Pool but remain withdraw capability - even though no deposit is expected to happen on this pool prior to the upgrade).

**Review by lebascou — APPROVED** [review]

**Pull request merged**